service/dap/server.go

```diff
@@ -1135,8 +1133,16 @@ func (s *Server) onVariablesRequest(request *dap.VariablesRequest) {
 				cfqname = "" // complex children are not struct fields and can't be accessed directly
 			}
 			cvalue, cvarref := s.convertVariable(c, cfqname)
+
+			// Annotate any shadowed variables to "(name)" in order
+			// to distinguish from non-shadowed variables.
+			name := c.Name
+			if c.Flags&proc.VariableShadowed == proc.VariableShadowed {
+				name = fmt.Sprintf("(%s)", name)
+			}
+
 			children[i] = dap.Variable{
-				Name:               c.Name,
+				Name:               name,
 				EvaluateName:       cfqname,
```

Comment:
doesn't EvaluateName need change too?

@aarzilli @derekparker we noticed that accessing shadowed variables (printing/setting) from dlv cli doesn't work. It looks like we can see them only by `locals`
```
(dlv) locals
(animal) = "dog"
animal = "cat"
(dlv) p (animal)
"cat"
```

Is it intended? Should we return an error if users try to evaluate scoped variables?

Reply:
I intentionally didn't change evaluate name in this PR. Adding parens does not actually affect the expression that is evaluated so it doesn't change it, so I thought keeping evaluateName the same will be clearer that the expression that will be evaluated is just "animal". If "(animal)" goes back to being not shadowed, then its name becomes "animal" again anyway.

I considered removing evaluateName, which in the vscode UI removes the 'Add to Watch' option, but a user may want to watch the variable when it is in scope later, so thought it best to keep the evaluateName without modification.

Reply:
> we noticed that accessing shadowed variables (printing/setting) from dlv cli doesn't work

It's intentional, the argument of print is an expression that's evaluated similarly to the way Go does, so parenthesis do nothing. We could invent some way to access shadowed variables inside expressions but nobody ever asked for it before now.

Reply:
I don't think it's urgent and a TODO for now would work just fine, but I can't we use the magic scopeless syntax for shadowed variables and use that as evaluateName? Like here:
https://github.com/go-delve/delve/blob/8b206092276a50aaa7edd682fa22d95ca9ba8070/service/dap/server.go#L1220-L1222
or here
https://github.com/go-delve/delve/blob/8b206092276a50aaa7edd682fa22d95ca9ba8070/service/dap/server.go#L1056

Reply:
One problem of allowing users to add the address-based expression to the watch option is the watch expressions are preserved across multiple debug sessions and become stale. I think not allowing users to access the shadowed variables but evaluate only within the current scope is intuitive.

Reply:
It isn't guaranteed to work, the variable could have a fake address due to having been partially registerized. This will become more apparent after the regabi change.

Reply:
That makes sense. That's why Add to Watch expressions are meaningful and user-friendly fully qualified names and not address-based expressions. Staying consistent with this would mean not being able to evaluate shadowed variables. Agree that it's intuitive.

What does "partially registered" mean? There is an exception to the above for complex map keys and corresponding value because keys don't have their own names. Would this be an issue for those?

Reply:
> What does "partially registered" mean?

Partially stored in a CPU register, instead of being in memory.

> There is an exception to the above for complex map keys and corresponding value because keys don't have their own names. Would this be an issue for those?

No.